state_transfer/load_balancer_endpoint.go

```diff
@@ -0,0 +1,84 @@
+package state_transfer
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type LoadBalancerEndpoint struct {
+	hostname string
+	port     int32
+}
+
+func (r *LoadBalancerEndpoint) createEndpointResources(c client.Client, t Transfer) error {
+	err := createLoadBalancerService(c, r, t)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createLoadBalancerService(c client.Client, r *LoadBalancerEndpoint, t Transfer) error {
+	serviceSelector := labels
+	serviceSelector["pvc"] = t.PVC().Name
+
+	service := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      t.PVC().Name,
+			Namespace: t.PVC().Namespace,
+			Labels:    labels,
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name:       t.PVC().Name,
+					Protocol:   v1.ProtocolTCP,
+					Port:       t.Transport().Port(),
+					TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: t.Transport().Port()},
+				},
+			},
+			Selector: serviceSelector,
+			Type:     v1.ServiceTypeLoadBalancer,
+		},
+	}
+
+	err := c.Create(context.TODO(), &service, &client.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	//FIXME. Seems to take a moment, probably something better to do than wait 5 seconds
```

Comment:
you can use a poller, it will be slightly better. https://github.com/kubernetes/kubernetes/blob/2453f07e933a6d2ac0be1eb35061a2154ccce5b8/test/e2e/framework/util.go#L269